### ISSUE TITLE
🐛 Bug Report: `yarn backend:build` doesn't forward config options on to app build

### ISSUE THREAD

**Issue by afscrome** [backstage/backstage#15274]
### 📜 Description

When running `yarn backend:build --config foo.yaml` when teh app is being bundled into the backend, the backend ignores the `--config` option provided and continues to use the default config file

### 👍 Expected behavior

The `--config` arguments should be passed on to the app build. 

### 👎 Actual Behavior with Screenshots

The app build continues to just use `app-config.yaml`

```
yarn build:backend --config foo.yaml
yarn run v1.22.19
$ yarn workspace backend build --config foo.yaml
$ backstage-cli package build --config foo.yaml
Building app separately because it is a bundled package

app:  
app:  $ backstage-cli package build
app:  
app:  Loaded config from app-config.yaml
app:  
```

### 👟 Reproduction steps

1. Run `yarn backend:build --config foo.yaml`

### 📃 Provide the context for the Bug.

I am upgrading from 1.7.0 to 1.8.3 and in the process migrating from `yarn build` to `yarn build:backend`.

I'm able to work around this by updating the build script in `packages/app/package.json` (As suggested at https://discord.com/channels/687207715902193673/1026835211285184593 )
```json
    "build": "backstage-cli package build --config ../../app-config.yaml --config ../../app-config.production.yaml",
```

### 🖥️ Your Environment

OS:   Linux 5.15.79.1-microsoft-standard-WSL2 - linux/x64
node: v16.14.0
yarn: 1.22.19
cli:  0.21.1 (installed)
backstage:  1.8.3

Dependencies:
  @backstage/app-defaults                          1.0.8
  @backstage/backend-common                        0.16.0
  @backstage/backend-plugin-api                    0.1.4
  @backstage/backend-tasks                         0.3.7
  @backstage/catalog-client                        1.1.2
  @backstage/catalog-model                         1.1.3
  @backstage/cli-common                            0.1.10
  @backstage/cli                                   0.21.1
  @backstage/config-loader                         1.1.6
  @backstage/config                                1.0.4
  @backstage/core-app-api                          1.2.0
  @backstage/core-components                       0.12.0
  @backstage/core-plugin-api                       1.1.0
  @backstage/dev-utils                             1.0.8
  @backstage/errors                                1.1.3
  @backstage/integration-react                     1.1.6
  @backstage/integration                           1.4.0
  @backstage/plugin-api-docs                       0.8.11
  @backstage/plugin-app-backend                    0.3.38
  @backstage/plugin-auth-backend                   0.17.1
  @backstage/plugin-auth-node                      0.2.7
  @backstage/plugin-azure-devops-backend           0.3.17
  @backstage/plugin-azure-devops-common            0.3.0
  @backstage/plugin-azure-devops                   0.2.2
  @backstage/plugin-catalog-backend-module-azure   0.1.9
  @backstage/plugin-catalog-backend-module-github  0.2.1
  @backstage/plugin-catalog-backend-module-msgraph 0.4.4
  @backstage/plugin-catalog-backend                1.5.1
  @backstage/plugin-catalog-common                 1.0.8
  @backstage/plugin-catalog-graph                  0.2.23
  @backstage/plugin-catalog-import                 0.9.1
  @backstage/plugin-catalog-node                   1.2.1
  @backstage/plugin-catalog-react                  1.2.1
  @backstage/plugin-catalog                        1.6.1
  @backstage/plugin-explore-react                  0.0.23
  @backstage/plugin-explore                        0.3.42
  @backstage/plugin-github-actions                 0.5.11
  @backstage/plugin-github-pull-requests-board     0.1.5
  @backstage/plugin-home                           0.4.27
  @backstage/plugin-org                            0.6.1
  @backstage/plugin-permission-common              0.7.1
  @backstage/plugin-permission-node                0.7.1
  @backstage/plugin-permission-react               0.4.7
  @backstage/plugin-proxy-backend                  0.2.32
  @backstage/plugin-scaffolder-backend             1.8.0
  @backstage/plugin-scaffolder-common              1.2.2
  @backstage/plugin-scaffolder                     1.8.0
  @backstage/plugin-search-backend-module-pg       0.4.2
  @backstage/plugin-search-backend-node            1.0.4
  @backstage/plugin-search-backend                 1.1.1
  @backstage/plugin-search-common                  1.1.1
  @backstage/plugin-search-react                   1.2.1
  @backstage/plugin-search                         1.0.4
  @backstage/plugin-stack-overflow                 0.1.7
  @backstage/plugin-tech-radar                     0.5.18
  @backstage/plugin-techdocs-backend               1.4.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.6
  @backstage/plugin-techdocs-node                  1.4.2
  @backstage/plugin-techdocs-react                 1.0.6
  @backstage/plugin-techdocs                       1.4.0
  @backstage/plugin-user-settings                  0.5.1
  @backstage/release-manifests                     0.0.7
  @backstage/test-utils                            1.2.2
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.1
  @backstage/version-bridge                        1.0.2
Done in 0.60s.

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, but I'm happy to collaborate on a PR with someone else

**Comment by Rugvip** [comment]
This generally isn't needed, since the `app-backend` plugin is will inject configuration at runtime anyway. If you need config for templating the `index.html` then that's a bit of an issue at the moment, but to be honest I'm feeling that the best solution is to try to move away from templating in `index.html`.

This might be something we could support still though, if someone wants to give it a shot we can see if makes sense.

**Comment by Pike** [comment]
I just came across this, and I was rather surprised that the FE config wasn't what I passed in to build:backend.

I think there are two problems here: 

One is that `app-config.local.yaml` is picked up by the build. That's unfortunate as it's really hard to build a docker image locally that matches that on CI.

The other is that templating in index.html can't access configuration in `app-config.production.yaml`. I tried to put our GA config into that, and broke prod ;-) 

We would need to bring the configs down to https://github.com/backstage/backstage/blob/a315fc68c01b1a54dd18b9ad27626d1fe5e86b26/packages/cli/src/lib/packager/createDistWorkspace.ts#L193-L205, and probably only for backstage-cli build variants, too?
